### PR TITLE
[0.2.0] Lockup: Replace access keys with external account

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
### `0.2.0`

- Replaced owner's access keys with the owner's account. The access is now controlled through the predecessor account ID similar to NEAR foundation access.
  This allows to be more flexible with the account access including multi-sig implementation.
- The lockup contract account should not have any access keys until the account is fully vested and unlocked.
  Only then the owner can add the full access key.
- Removed methods for adding and removing staking/main access keys.
- Added a view method to get the account ID of the owner.

Fixes: #31

### CODE REVIEW

File: lockup/src/internal.rs [hunk 1]
```diff
@@ -97,4 +97,12 @@ impl LockupContract {
             env::panic(b"No NEAR Foundation account is specified in the contract");
         }
     }
+
+    pub fn assert_owner(&self) {
```
Comment (fckt) [inline lockup/src/internal.rs:101]: Maybe it makes sense to have `assert_foundation`, by analogy.

Comment (willemneal) [reply lockup/src/internal.rs:101]: There `assert_called_by_foundation` above.

**Comment by evgenykuzyakov** [comment]
@fckt I don't think we can enforce that the owner account exists or have access. Checking the validity of the owner account should be enough.

**Comment by evgenykuzyakov** [comment]
> Do you think there should be a test for a non-owner account calling an owner method?

There is a unit test for this.

**Comment by fckt** [comment]
I mostly mean that we need a deployment script to make sure that deployer (at least) didn't made a mistake

**Comment by willemneal** [comment]
> I mostly mean that we need a deployment script to make sure that deployer (at least) didn't make a mistake on initialization.

Yeah that would be good, but I think that is covered in the main README in the repo.

**Comment by fckt** [comment]
Ok, it's not a blocker. I just think it's it's important to have a "deployer" script which takes all the necessary parameters and then creates accounts, deploys contracts and does asserts.

**Comment by evgenykuzyakov** [comment]
> Ok, it's not a blocker. I just think it's it's important to have a "deployer" script which takes all the necessary parameters and then creates accounts, deploys contracts and does asserts.

Makes sense. I planned to work on this, so we should include it into this repo

**Pull request merged**